src/common/utils/utils.service.ts

```diff
@@ -90,25 +91,31 @@ export class UtilsService {
     if (!url) {
       return {};
     }
-
     url = url.slice(url.indexOf('?') + 1);
 
     let params = JSON.parse(`{"${url.replace(/&/g, '\",\"').replace(/=/g, '\":\"')}"}`);
-
+    params = reduce(params, (result: UrlParameters, value: string, key: string) => {
+      result[key] = decodeURI(value);
+      return result;
+    }, {});
     if (params.regions) {
       params.regions = params.regions.split(',');
     }
 
     if (params.countries) {
       params.countries = params.countries.split(',');
     }
-
     return params;
   }
 
-  public objToQuery(data: any): string {
-    return Object.keys(data).map((k: string) => {
-      return encodeURIComponent(k) + '=' + data[k];
-    }).join('&');
+  public objToQuery(data: UrlParameters): string {
+
+    return reduce(data as NumericDictionary<{}>, (result, value, key) => {
```

Comment:
@Miffael why NumericDictionary if UrlParameters it's an object with strings values?